openedx/openedx-platform
# fix: remove the "Legacy Library" button if libraries v1 are disabled [FC-0062]

## Description

Removes the "Legacy Library" button from the legacy Studio "new block" button array if Libraries v1 are disabled, either via waffle flag or via the `ENABLE_CONTENT_LIBRARIES` feature flag.

All libraries enabled (default):

![all_libraries_enabled](https://github.com/user-attachments/assets/15774730-19c0-496b-8f5d-5da5f31b32de)

Libraries v1 only enabled: 

![v1_libraries_only](https://github.com/user-attachments/assets/f9d9f5c2-f23f-4e5f-9035-82fd5ec63c26)

Libraries v2 only enabled:

![v2_libraries_only](https://github.com/user-attachments/assets/4899b680-c730-43df-a31a-ff1bd0d35961)

No libraries enabled:

![no_libraries_enabled](https://github.com/user-attachments/assets/59e23ece-1d1b-41c3-9928-a09c209ac34c)

## Supporting information

Addresses: https://github.com/openedx/modular-learning/issues/237
Private-ref: [FAL-3855](https://tasks.opencraft.com/browse/FAL-3855)

## Testing instructions

1. Disable the `contentstore.new_studio_mfe.use_new_unit_page` Waffle flag to access the legacy Studio Unit page for a course.
2. Change the settings for the `contentstore.new_studio_mfe.disable_new_libraries` (v2 libraries) and `contentstore.new_studio_mfe.disable_legacy_libraries` (v1 libraries) waffle flags to toggle support for the various libraries content buttons.
   If `contentstore.new_studio_mfe.disable_new_libraries` is disabled, the Libraries v2 buttons for "Library Content (beta)" and "Problem Bank (beta)" should be shown.
   If `	contentstore.new_studio_mfe.disable_legacy_libraries` is disabled, the "Legacy Library" button should be shown.
3. Change `settings.FEATURES['ENABLE_CONTENT_LIBRARIES'] = False`
   This disables both libraries v1 + v2, so all 3 buttons should be removed, regardless of the relevant waffle flags.

## Deadline

ASAP
